lib/pathutils.c

```diff
@@ -92,3 +94,108 @@ get_filename_extension(const gchar *filename)
 
   return start_ext + 1;
 }
+
+static gchar**
+_add_path_to_filenames(gchar *path, gchar **filenames, size_t *pos, size_t *array_len)
+{
+  if (*pos < *array_len)
+    {
+      filenames[*pos] = path;
+      (*pos)++;
+    }
+  else
+    {
+      *array_len *= 2;
```

Comment:
Is it possible to use some glib data structure? Like `GPtrArray`? Then you would not need to implement such logics.

Reply:
Thanks, it helped a lot! :)